qadence/types.py

```
@@ -180,7 +180,7 @@ class AnsatzType(StrEnum):
     """Alternating Layer Ansatz."""
```

Shouldn't there be an AGPSR diff mode?

---

Well, that's up for debate. Currently I decided not to introduce a separate diff mode since it's very similar in everything to GSPR. Essentially, you pass `n_eqs` and you use aGPSR.

---

I see, maybe an easy option could be to have the `n_eqs` added to the possible model configs, and have it by default set to use the full GPSR.

---

Similarly to how you can change the solver method for the time-dependent blocks.

---

Hm, I'd rather not silently fall back onto smtg a user might not _explicitly_ asked for if I'm totally honest.

---

Agreed but if I am understanding the code correctly, and since there is no explicit `aGPSR` diff mode, it seems like this would be an ok usage of the model config. If the user asks for `GPSR` they get the full `GPSR`, and they want the approximate one, they change the model config with the number of equations they would like to use.

---

Alright but it has to be clearly documented.

---

@RolandMacDoland, what do you mean?